src/components/VarLoader.tsx

```diff
@@ -48,79 +53,97 @@ export default class VarLoader extends React.Component<
     });
   }
 
-  /**
-   *
-   * @param variables An array of Variable objects to display in the loader modal
-   */
-  setVariables(variables: Array<Variable>) {
-    if (this.state.loadedVariables.length > 0) {
-      let newVariables = new Array<Variable>();
-      variables.forEach((newVariable: Variable, newVarIndex: number) => {
-        let found = false;
-        this.state.loadedVariables.forEach((loadedVariable: Variable) => {
-          if (newVariable.name == loadedVariable.name) {
-            //newVariables.push(loadedVariable);
-            found = true;
-          }
-        });
-        if (!found) {
-          newVariables.push(newVariable);
+  isSelected(varName: string) {
+    return this.state.selectedVariables.indexOf(varName) >= 0;
+  }
+
+  async loadSelectedVariables() {
+    // Once the load button is clicked, load only the files that were selected.
+    try {
+      this.state.fileVariables.forEach(async (variable: Variable) => {
+        let ind = this.state.selectedVariables.indexOf(variable.cdmsID);
+        if (ind >= 0) {
+          console.log(`Added: ${variable.name}`);
+          await this.props.loadFileVariable(variable);
+        } else {
+          console.log(`Skipped: ${variable.cdmsID}`);
         }
       });
-      this.setState({
-        variables: newVariables
-      });
-    } else {
-      this.setState({
-        variables: variables
-      });
+      // Select only the max number of slabs allowed for plot injection
+      let selection = this.state.selectedVariables;
+      if (selection.length > MAX_SLABS) {
+        selection = selection.slice(0, MAX_SLABS);
+      }
+      // Update the main widget's current selected variables
+      await this.props.updateSelectedVariables(selection);
+      // Reset the selected files in the var loader when done
+      this.setState({ selectedVariables: new Array<string>() });
+    } catch (error) {
+      console.log(error);
```

Comment:
remove

Reply:
I wasn't sure if you meant for me to remove the console.log, or the whole try-catch. Since the try catch needs to do something and you mentioned they're not necessary if they don't handle the error, I just removed the whole try-catch thing.